kafka-producer-java/src/main/java/NoneBlockingProducer.java

```diff
@@ -0,0 +1,23 @@
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+
+public class NoneBlockingProducer extends Producer {
+
+    NoneBlockingProducer(Config config, Monitor monitor) {
+        super(config, monitor);
+    }
+
+    @Override
+    public boolean produce(ProducerRequest producerRequest) {
+        var executionStart = (new Date()).getTime();
+        try {
+            kafkaProducer.send(createRecord(producerRequest, executionStart)).get();
+            ready = true;
+            Monitor.produceSuccess(producerRequest, executionStart);
+        } catch (InterruptedException | ExecutionException e) {
+            ready = false;
+            Monitor.produceError(e);
+        }
+        return true;
```

Comment:
return true, only if succeeded. also, I'm dont think we need to hide the error, you can rethrow it as it is

Reply:
the rethrow makes the stream code more complicated. I prefer not to rethrow